driver-core/src/main/com/mongodb/operation/CommitTransactionOperation.java

```diff
@@ -135,8 +142,16 @@ public BsonDocument create(final ServerDescription serverDescription, final Conn
                 }
```

Comment:
I'm not clear on why the recoveryToken isn't also added when already alreadyCommitted is true.

Reply:
The recovery token is added in `getRetryCommandModifier`.

Reply:
OK, it makes sense now.